src/xrpld/ledger/detail/View.cpp

```diff
@@ -1153,18 +1153,22 @@ addEmptyHolding(
     auto const& dstId = accountID;
     auto const high = srcId > dstId;
     auto const index = keylet::line(srcId, dstId, currency);
-    auto const sle = view.peek(keylet::account(accountID));
-    if (!sle)
-        return tefINTERNAL;
-    if ((sle->getFlags() & lsfDefaultRipple) == 0)
-        return tecINTERNAL;
+    auto const sleSrc = view.peek(keylet::account(srcId));
+    auto const sleDst = view.peek(keylet::account(dstId));
+    if (!sleDst || !sleSrc)
+        return tefINTERNAL;  // LCOV_EXCL_LINE
+    if (!sleSrc->isFlag(lsfDefaultRipple))
+        return tecINTERNAL;  // LCOV_EXCL_LINE
```

Comment:
I know this was my suggestion, but it occurred to me that these are not good return codes unless we've already checked these conditions beforehand. Specifically, it is perfectly legal to have a trust line to an "issuer" that doesn't have default ripple set, thought we probably don't want to allow a vault to be created using that issuer, because you wouldn't be able to make a deposit. I suggest a check in `preclaim` for an transactor that calls `addEmptyHolding`, which returns `tecNO_ISSUER` if the issuer is not found, and `terNO_RIPPLE` if the issuer doesn't have `lsfDefaultRipple`. (They could turn it on later, making the transaction valid. Also, there's no `tec` code that I can find that fits.)

Reply:
`terNO_ACCOUNT` in case if there's no SLE object for the issuer seem more consistent with existing older transactions.